### ISSUE TITLE
java.util.NoSuchElementException in LoyaltyCardEditActivity ChooseCardImage onClick

### ISSUE THREAD

**Issue by TheLastProject** [CatimaLoyalty/Android#1211]
Happens if someone disables their camera app and clicks "Take a photo" in the Photos section of "Edit Card" or if the user clicks "Take a photo" in the Set icon menu of the "Main" section.

The user in question tried this 3 times in a row. I'm not sure why they expected to be able to take a photo with their camera app disabled.

Labelling with severity critical because this is a crash and Catima should never crash.

Device: Unknown
Android: 11 (SDK 30)

```
Exception java.util.NoSuchElementException: No Activity found to handle Intent { act=android.media.action.IMAGE_CAPTURE flg=0x3 clip={text/uri-list {...}} (has extras) }
  at protect.card_locker.LoyaltyCardEditActivity$ChooseCardImage.lambda$onClick$4 (LoyaltyCardEditActivity.java:1274)
  at protect.card_locker.LoyaltyCardEditActivity$ChooseCardImage.$r8$lambda$Gr6vI1wx4hx6kYKE1KKL-3bjBVo
  at protect.card_locker.LoyaltyCardEditActivity$ChooseCardImage$$ExternalSyntheticLambda0.onClick
  at androidx.appcompat.app.AlertController$AlertParams$3.onItemClick (AlertController.java:1068)
  at android.widget.AdapterView.performItemClick (AdapterView.java:330)
  at android.widget.AbsListView.performItemClick (AbsListView.java:1210)
  at android.widget.AbsListView$PerformClick.run (AbsListView.java:3202)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at android.os.Looper.loop (Looper.java:223)
  at android.app.ActivityThread.main (ActivityThread.java:7705)
  at java.lang.reflect.Method.invoke
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:592)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:952)
```

**Comment by TheLastProject** [comment]
Sure, contributions are always welcome :)

**Comment by TheLastProject** [comment]
Fixed in 65acb88

While #1287 helped, there were some cases where it still crashed. After testing a bit more, it seemed that the catch was correct, the location in the code just wasn't covering all possible paths, causing one crash to remain on `at protect.card_locker.LoyaltyCardEditActivity.takePhotoForCard(LoyaltyCardEditActivity.java:1132)`

```
FATAL EXCEPTION: main
Process: me.hackerchick.catima.debug, PID: 16952
java.lang.RuntimeException: Failure delivering result ResultInfo{who=@android:requestPermissions:, request=100, result=-1, data=Intent { act=android.content.pm.action.REQUEST_PERMISSIONS (has extras) }} to activity {me.hackerchick.catima.debug/protect.card_locker.LoyaltyCardEditActivity}: android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.media.action.IMAGE_CAPTURE flg=0x3 clip={text/uri-list hasLabel(0) {U(content)}} (has extras) }
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5335)
	at android.app.ActivityThread.handleSendResult(ActivityThread.java:5374)
	at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:67)
	at android.app.servertransaction.ActivityTransactionItem.execute(ActivityTransactionItem.java:45)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:138)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2303)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loopOnce(Looper.java:201)
	at android.os.Looper.loop(Looper.java:288)
	at android.app.ActivityThread.main(ActivityThread.java:7884)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
Caused by: android.content.ActivityNotFoundException: No Activity found to handle Intent { act=android.media.action.IMAGE_CAPTURE flg=0x3 clip={text/uri-list hasLabel(0) {U(content)}} (has extras) }
	at android.app.Instrumentation.checkStartActivityResult(Instrumentation.java:2200)
	at android.app.Instrumentation.execStartActivity(Instrumentation.java:1839)
	at android.app.Activity.startActivityForResult(Activity.java:5506)
	at androidx.activity.ComponentActivity.startActivityForResult(ComponentActivity.java:728)
	at androidx.core.app.ActivityCompat$Api16Impl.startActivityForResult(ActivityCompat.java:809)
	at androidx.core.app.ActivityCompat.startActivityForResult(ActivityCompat.java:246)
	at androidx.activity.ComponentActivity$2.onLaunch(ComponentActivity.java:243)
	at androidx.activity.result.ActivityResultRegistry$2.launch(ActivityResultRegistry.java:175)
	at androidx.activity.result.ActivityResultLauncher.launch(ActivityResultLauncher.java:47)
	at protect.card_locker.LoyaltyCardEditActivity.takePhotoForCard(LoyaltyCardEditActivity.java:1132)
	at protect.card_locker.LoyaltyCardEditActivity.onMockedRequestPermissionsResult(LoyaltyCardEditActivity.java:1020)
	at protect.card_locker.LoyaltyCardEditActivity.onRequestPermissionsResult(LoyaltyCardEditActivity.java:1011)
	at android.app.Activity.dispatchRequestPermissionsResult(Activity.java:8810)
	at android.app.Activity.dispatchActivityResult(Activity.java:8668)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5328)
	... 13 more
```

**Comment by LuiscVillarrealL** [comment]
happy to help!
